lib/posix-mmap/Config.uk

```diff
@@ -2,7 +2,6 @@ menuconfig LIBPOSIX_MMAP
 	bool "posix-mmap: Memory mapping related functions"
 	default n
 	depends on LIBUKVMEM
-	depends on !LIBUKMMAP
```

Comment:
Why change this to be the other way around? (have `ukmmap` depend on `!LIBPOSIX_MMAP` instead of `posix-mmap` depend on `!LIBUKMMAP`)?

Reply:
@StefanJum, the reason is that most of the external libraries don't have mentions of `LIBPOSIX_MMAP`, but rather `LIBUKMMAP`. 

For example, without this commit, if you first select `compiler-rt` in the menuconfig, `ukmmap` gets selected because the library's `Config.uk` file includes the statement `imply UKMMAP`. If you afterwards select `libgo`, which has the `select LIBPOSIX_MMAP` requirement, it would indeed select `posix-mmap`, but for some reason it also keeps `ukmmap` selected as well, which leads to a double definition conflict.

The dependency switch that I've introduced prevents that from happening, but I do not have a perfect explanation for this behavior, unfortunately.

Reply:
Hum... one additional thing to do in order to avoid the problem is by patching `compiler_rt` with
`imply LIB_UKMMAP if !LIBPOSIX_MMAP`.
Additionally, we probably should add a warning or error with `support/scripts/configupdate` as soon as ukmmap and posix_mmap are both. I think this would be a reasonable workaround as long as `ukmmap` exists. Longer term the plan was to migrate the left functionality from `ukmmap` to `posix-mmap` anyways.

Overall, I will approve this PR and will open an issue for updating the `configupdate` script.